packages/datetime/src/dateInput.tsx

```diff
@@ -343,6 +343,10 @@ export class DateInput extends AbstractComponent<IDateInputProps, IDateInputStat
      * This is a no-op unless moment-timezone's setDefault has been called.
      */
     private fromDateToMoment = (date: Date) => {
+        if (null == date) {
```

Comment:
oh wait nvm it's not that simple. see the docs for the props. `null` and `undefined` actually mean different things here.

> To display no date in the input field, pass `null` to the value prop.